src/borg/helpers.py

```diff
@@ -231,10 +231,13 @@ def list(self, sort_by=(), reverse=False, glob=None, first=None, last=None):
         archives = [x for x in self.values() if regex.match(x.name) is not None]
         for sortkey in reversed(sort_by):
             archives.sort(key=attrgetter(sortkey))
-        if reverse or last:
+        if reverse:
             archives.reverse()
-        n = first or last or len(archives)
-        return archives[:n]
+        if first:
+            archives = archives[:first]
+        elif last:
+            archives = archives[max(len(archives) - last, 0):]
+        return archives
```

Comment:
Yeah, the only ordering that actually makes sense is match-sort-firstlast-reverse.